### ISSUE TITLE
Path D: Tail-aware extension — extreme value component for Tobit loss

### ISSUE THREAD

**Issue by Polichinel** [views-platform/views-hydranet#38]
## Sequence Position

> **This is Step 3a of 6** in the zero-inflation instability remediation sequence: **A → E → D/B → C/F**
>
> **Depends on:** #36 (Path A: Tobit loss) — must be validated first.
> **Parallel with:** Path B (Heckman coupling) — these two address different calibration gaps and can be evaluated independently.
>
> This is a **refinement** issue, not a stability fix. Only pursue after Path A proves stable and Path E addresses long-horizon quality. The question this issue answers: "Are we systematically underpredicting extreme conflict events?"

---

### The tail problem

The Tobit censored-normal likelihood (Path A, #36) assumes the latent intensity z* follows a normal distribution. Conflict fatalities do not — they have a heavy right tail. A few grid cells in any given month have very high fatality counts (South Sudan, Syria, Yemen), while the vast majority have zero or near-zero.

Wilson et al. (KDD 2022) demonstrate empirically that simple hurdle and censoring models **underpredict the frequency of extreme events** in zero-inflated heavy-tailed spatiotemporal data. Their DEMM (Deep Extended Mixture Model) shows that adding a Generalized Pareto (GP) component specifically for the tail substantially improves extreme event calibration.

### Our current tail management

The existing `qs99` regularizer (`qs99_weight=0.1`, `qs99_tau=0.99`) adds a quantile-score penalty for the 99th percentile. This is an ad hoc approach — it penalizes tail misprediction but doesn't model the tail distribution. The DEMM framework suggests a principled alternative.

### Why this matters for our domain

In conflict forecasting, extreme events are the most policy-relevant predictions. A model that correctly predicts "low activity" in 95% of cells but misses the 5 cells with mass casualty events has limited operational value. Tail calibration directly affects the model's utility for early warning.

---

## Literature Support

| Paper | Authors | Year | Key finding | Link |
|-------|---------|------|-------------|------|
| DEMM | Wilson, Gao, Bhatt, Flaxman | 2022 | Three-component mixture (Bernoulli + truncated log-normal + Generalized Pareto) outperforms hurdle models on zero-inflated heavy-tailed data. Hurdle alone underpredicts extreme event frequency. Variable threshold training improves robustness. | KDD 2022 |
| DynAttn | Iacus et al. | 2025 | ZINB likelihood partially handles overdispersion through the NB component, but doesn't explicitly model heavy tails. | [arXiv](https://arxiv.org/abs/2512.21435) |
| TOBART | O'Neill | 2024 | TOBART-NP uses Dirichlet process mixture errors, which can capture heavy tails nonparametrically. Outperforms fixed-distribution variants. | [arXiv](https://arxiv.org/abs/2211.07506) |

---

### Option 1: Retain `qs99` as pragmatic tail penalty alongside Tobit

**What:** Keep the existing `qs99_weight` / `qs99_tau` regularizer. The Tobit loss handles zero cells; `qs99` penalizes tail misprediction. They're additive.

```python
total_loss = tobit_loss(pred, target) + qs99_weight * quantile_score_99(pred, target)
```

**Pros:** Zero implementation effort. Already tested (in S3 isolation).
**Cons:** Ad hoc. Doesn't model the tail distribution — just penalizes deviations. No theoretical justification for the weight or τ choice.

**When to use:** If S3 (qs99 isolation) passes and tail calibration is acceptable.

### Option 2: Generalized Pareto tail component (DEMM-inspired)

**What:** For observations above a threshold u, model the exceedance (y - u) with a Generalized Pareto distribution:

```
P(Y - u > y | Y > u) = (1 + ξy/σ_GP)^{-1/ξ}
```

The full loss becomes a mixture:
```
L(y, μ, σ, ξ, σ_GP) = {
    -log Φ(-μ/σ)                                        if y = 0     (censored)
    -log φ((y - μ)/σ) + log σ                           if 0 < y ≤ u (body)
    -log GP(y - u; ξ, σ_GP)                              if y > u     (tail)
}
```

**Pros:** Principled tail modeling. GP distribution is the theoretical limiting distribution for exceedances (Pickands–Balkema–de Haan theorem). Separates body and tail calibration.
**Cons:** Three additional parameters (u, ξ, σ_GP). Threshold u selection is notoriously difficult in extreme value theory. Adds complexity to an already-novel loss function.

**When to use:** If Path A (Tobit) is stable but evaluation shows systematic underprediction of high-fatality cells.

### Option 3: Dirichlet process mixture errors (TOBART-NP inspired)

**What:** Instead of assuming normal errors in the Tobit likelihood, use a nonparametric mixture of normals to capture the error distribution. This adapts to heavy tails without explicitly specifying a tail distribution.

**Pros:** Theoretically elegant. Adapts to whatever the true error distribution is.
**Cons:** Significantly more complex. MCMC-based inference in TOBART; unclear how to do this efficiently in a deep learning setting. Likely requires variational approximation.

**When to use:** Research exploration only. Not a near-term option.

---

## Recommended Approach

**Start with Option 1** (retain qs99). It's already implemented and testable. After Path A is validated, evaluate tail calibration:

1. Run the Tobit model (Path A) through evaluation
2. Compute per-cell error distributions, stratified by true value magnitude
3. Plot calibration curves for the upper quantiles (90th, 95th, 99th percentile)
4. If systematic underprediction of extreme events: implement Option 2 (GP tail)
5. If qs99 provides adequate tail penalty: keep it

---

## Diagnostic Criteria

"Do we have a tail problem?" can be assessed after Path A is running:

- **CRPS decomposition:** If the CRPS is dominated by the tail component (large positive values), the Tobit normal assumption is insufficient.
- **MCR at high quantiles:** If MCR (Mean Calibration Ratio) for active cells is >3× baseline, predictions are biased in the tail.
- **Visual inspection:** Plot predicted vs actual for the top 1% of cells. Systematic underestimation = tail problem.

---

### Phase 1: Threshold selection

Choose the threshold u that separates "body" from "tail." Options:
- Fixed percentile of positive values (e.g., 95th percentile of y | y > 0)
- Variable threshold as in DEMM (Wilson et al. 2022): train with multiple thresholds, select by validation

### Phase 2: GP loss component

```python
class GPTailLoss(nn.Module):
    def __init__(self, threshold, xi_init=0.1, sigma_init=1.0):
        super().__init__()
        self.threshold = threshold
        self.log_sigma = nn.Parameter(torch.tensor(math.log(sigma_init)))
        self.xi = nn.Parameter(torch.tensor(xi_init))

    def forward(self, pred, target):
        exceedance_mask = target > self.threshold
        if not exceedance_mask.any():
            return torch.tensor(0.0, device=pred.device)
        excess = target[exceedance_mask] - self.threshold
        sigma = self.log_sigma.exp()
        # GP log-likelihood
        z = excess / sigma
        if self.xi.abs() > 1e-6:
            log_prob = -torch.log(sigma) - (1 + 1/self.xi) * torch.log1p(self.xi * z)
        else:
            log_prob = -torch.log(sigma) - z  # exponential limit
        return -log_prob.mean()
```

### Phase 3: Combined Tobit + GP loss

```python
class TobitGPLoss(nn.Module):
    def forward(self, pred, target):
        # Censored cells: Tobit
        # Body cells: Gaussian NLL
        # Tail cells: GP NLL
        ...
```

### Phase 4: Validation

Compare against baseline Tobit:
- CRPS overall (should not degrade)
- CRPS for top 1% of cells (should improve)
- MCR at 99th percentile (should decrease toward 1.0)

---

## Acceptance Criteria

- [ ] Tail calibration diagnostic completed after Path A validation
- [ ] Decision made: qs99 sufficient vs GP component needed
- [ ] If GP component: implemented, validated, integrated with Tobit loss
- [ ] If qs99 retained: documented rationale for keeping ad hoc approach
- [ ] All existing tests pass

**Comment by Polichinel** [comment]
**PARKED** — deferred fallback (not rejected); see ZINB epic #97 / checklist #104 / C-139. Revisit only if the ZINB head fails. Rationale: loss/calibration investigate-X — deferred; ZINB is the chosen direction
